src/middlewared/middlewared/plugins/system.py

```diff
@@ -453,6 +453,9 @@ def license_update(self, license):
             stdout=subprocess.PIPE,
         )).communicate())[0].decode().strip() or None
 
+        # https://superuser.com/questions/893560/how-do-i-tell-if-my-memory-is-ecc-or-non-ecc/893569#893569
+        ecc_results = re.findall(r'Total Width:\s*(\d*)', (await run(['dmidecode', '-t', '17'])).stdout.decode())
```

Comment:
Could you define/compile the regex at the top, please?

RE_MEMWIDTH = re.compile(r'Total Width:\s*(\d*)')

or something